### PR TITLE
Warn instead of error when service_worker is combined with background.scripts/page

### PR DESCRIPTION
**Pull request by willdurand**
Fixes https://github.com/mozilla/addons-linter/issues/5152

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/ADDLINT-423)

### CODE REVIEW

File: tests/unit/parsers/test.manifestjson.js [hunk 1]
```diff
@@ -3179,6 +3179,47 @@ describe('ManifestJSONParser', () => {
...
+          linter.collector,
+          {
+            io: {
+              files: { 'background_worker.js': '', 'bg.js': '', 'bg.html': '' },
+            },
+          }
+        );
+
+        expect(linter.collector.errors).toEqual([]);
+        // expect(manifestJSONParser.isValid).toEqual(true);
```
Comment (rpl) [inline tests/unit/parsers/test.manifestjson.js:3209]: @willdurand I believe you may have commented this while you were figuring out where the strict behavior related to the additional properties was coming from and forgot to uncomment it, it does currently pass as expected if uncommented.

would you mind to also move it after the assertion on the warnings along with uncommenting it? (in the past it was often annoying to get a failure from this assertion without having hit the more detailed assertion that follows it in this test and so I usually put this assertion at the end to make it easier to investigate regressions hit in ci jobs).

**Review by rpl — APPROVED** [review]
@willdurand lgtm, in the comments below just a nit for an assertion currently left as commented + some details about that schema update file history (at least the amount of history that I could gather without spending too much time on it).

**Comment by rpl** [comment]
@willdurand forgot to mention, there is a typo in the commit message (pesent -> present), I took care of fixing the one in the PR subject message derived from the commit message

**Review by Rob--W — APPROVED** [review]
LGTM, thanks for the fix.

Note: the commit title states that its purpose is to emit a warning. That's not the objective: instead, the goal is to get rid of the hard error. Downgrading the error to a warning meets that goal, so this patch is indeed a right fix.

I suggest to change the commit message to: `Warn instead of error when service_worker is combined with background.scripts/page`

**Pull request merged**